### PR TITLE
[WIP] added banner for mobile

### PR DESCRIPTION
**Pull request by zacel**
Added a banner for mobile with slightly different text. The text is google translated for en, no, nl, sk, tr, and it

* [ ]  .bd – Bangladesh (@redikod)
* [ ]  .ca – Canada (@mepa1363)
* [ ]  .dk – Denmark (@btutal)
* [x]  .en-AU – Australia (@bitbonsai)
* [x]  .en-IN – India
* [x]  .en-MT – Malta (@tkrunning)
* [x]  .en-NG – Nigeria (@desoga10)
* [x]  .en-SG – Singapore (@jinjie)
* [x]  .en-US – United States
* [x]  .en – United Kingdom (@amritnagi)
* [x]  .es-AR – Argentina (@waltergammarota)
* [x]  .es-CL – Chili (@rubentd)
* [x]  .es-CO – Colombia (@julianbetancourt)
* [x]  .es-ES – Spain (@jesgarciamartinez)
* [x]  .es-MX – Mexico (@alexalmadav) 
* [x]  .fr-FR – France (@bertrandlalo)
* [x]  .it – Italy (@zerbfra)
* [ ]  .ms-MY – Malaysia (@jinjie)
* [x]  .nl – The Netherlands (@adriaanvanrossum @Kilian h@NickolasBoyer @julianbetancourt)
* [x]  .no – Norway (@michaelmcmillan @fossecode)
* [ ]  .se – Sweden (@pontus @mihai-sysbio @MalinAhlberg @talavis)
* [x]  .sk – Slovakia (@sbocinec)
* [x]  .tr – Turkey (@btutal)
* [x]  .ua – Ukraine (@tarasmatsyk)
* [ ]  .pt-PT – Portugal (@andrezzoid @Painatalman)
* [ ]  .de-DE – Germany (@btutal @lseelenbinder)
* [ ]  .ch-DE – Switzerland (@lseelenbinder)

### CODE REVIEW

**Comment by zacel** [comment]
this is in reference to this issue https://github.com/BustByte/coronastatus/issues/328

**Comment by zacel** [comment]
photo_2020-03-25 20 02 12

File: app/locales/no.json [hunk 1]
```diff
@@ -49,6 +49,8 @@
   "NB!": "NB!",
   "This page is currently not affiliated with the authorities in any way.": "Denne siden er p.t. ikke tilknyttet FHI eller norske myndigheter på noen måte.",
   "Read how the data you enter is processed here!": "Les hvordan dataen du oppgir blir behandlet her!",
+  "This page is not affiliated with the authorities.":"Denne siden er ikke tilknyttet myndighetene.",
+  "Learn more!":"Lære mer!",
```
Comment (fossecode) [inline app/locales/no.json]: Les mer!

Comment (zacel) [reply app/locales/no.json]: I've changed it

**Comment by amritnagi** [comment]
Before including this, can we tweak margins a little bit so it's more balanced with the top/bottom margins.

Also, I'm thinking whether we should have a "Official Version Exists " flag in the config, and display an alternative message/link if there is an official government site?

**Comment by fossecode** [comment]
@amritnagi There is already a flag in the config for that, it is called `REDIRECT_TO_GOVERNMENT`.

This is how we display it now, but we should maybe add something to the banner as well.
image

**Comment by zacel** [comment]
> Before including this, can we tweak margins a little bit so it's more balanced with the top/bottom margins.
> 
> Also, I'm thinking whether we should have a "Official Version Exists " flag in the config, and display an alternative message/link if there is an official government site?

Can you elaborate on the requested margin change. Currently both mobile and desktop banner are the same banner.ejs so any changes will apply to both or there will be a banner-mobile.ejs to maintain as well. The margins look even to me in the screenshot above.

In reagrards to the link to the gov version, what about linking it when they click on the the text "This page is not associated with the authorities". I've tried different configurations of the banner and any more information on mobile leads to the banner dominating the screen.

**Comment by zacel** [comment]
@bertrandlalo @zerbfra @adriaanvanrossum @michaelmcmillan @btutal @alny @julianbetancourt @alexalmadav @sbocinec 

When you have a chance could you please paste translations/verify mine from google translate for these two lines. 

"This page is not affiliated with the authorities."
 
"Learn more!"

Thank you for any help

**Review by btutal — COMMENTED** [review]
I've just made a small change, rest looks fine for Turkish

**Comment by zacel** [comment]
thank you @btutal

**Comment by zerbfra** [comment]
We already have very similar strings in the translation file!

**Comment by zacel** [comment]
@zerbfra yes these are very similar to:
"This page is currently not affiliated with the authorities in any way."
and 
  "Read how the data you enter is processed here!"

The ones above are shortened version so they can fit on a mobile banner and convey the same meaning

**Comment by zerbfra** [comment]
Ok. So.
Learn more = “Scopri di più”

The other one should be: “Questo sito non è legato alle autorità italiane”

I’ll fix after on my fork if they don’t fit

**Comment by zacel** [comment]
> Ok. So.
> Learn more = “Scopri di più”
> 
> The other one should be: “Questo sito non è legato alle autorità italiane”
> 
> I’ll fix after on my fork if they don’t fit

thank you

**Comment by adriaandotcom** [comment]
Need help @zacel?

**Comment by zacel** [comment]
@adriaanvanrossum yeah I think so. I'm not super familiar with making pull requests so I'm unsure how to proceed. I've asked some of the other language experts to verify the translations and added the correct ones as I can. I'm unclear on what else I should do to prepare this branch to be merged? When I first created it there were no conflicts. I think I may have communicated the point of these translations poorly as some people say they are similar to existing ones, which is intentional as these are shortened messages to fit on a smaller screen. If you would be so kind the biggest help I would like is to be told what it is I'm supposed to do to move forward so I can learn that and be able to do it next time. Thank you for asking.

**Comment by adriaandotcom** [comment]
I fixed the merge conflicts for you, but there is one line that has two options:

"This page is currently not affiliated with the authorities in any way."

Should this be

1. "Táto stránka v súčasnosti nijako nesúvisí so slovenskými štátnymi orgánmi."
2. "Táto stránka je nezávislou iniciatívou dobrovoľníkov, nesúvisí so štátnými orgánmi."

Would you mind picking the one that's correct?

**Comment by zacel** [comment]
I don't speak Slovakian but according to google translate the second one seems to make more sense. I've changed the sk locale to use that one but wasn't able to find the duplicate line to remove it?

File: app/locales/sk.json [hunk 2]
```diff
@@ -48,6 +48,8 @@
   "Share on": "Zdieľajte na",
   "NB!": "Upozornenie!",
   "This page is currently not affiliated with the authorities in any way.": "Táto stránka je nezávislou iniciatívou dobrovoľníkov, nesúvisí so štátnými orgánmi.",
+  "This page is not affiliated with the authorities.":"Táto stránka nie je prepojená s orgánmi.",
```
Comment (sbocinec) [inline app/locales/sk.json]: "Táto stránka je nezávislou iniciatívou."

File: app/locales/sk.json [hunk 3]
```diff
@@ -48,6 +48,8 @@
...
+  "Learn more!":"Uč sa viac!",
```
Comment (sbocinec) [inline app/locales/sk.json]: "Zisti viac!"

**Review by sbocinec — APPROVED** [review]

**Comment by fossecode** [comment]
We should try to find a better way for gathering translations when new texts are added. These are all locales we have right now (many of them are english or spanish):
```
  | 'no'
  | 'nl'
  | 'en'
  | 'en-US'
  | 'en-AU'
  | 'en-MT'
  | 'es-MX'
  | 'sk'
  | 'it'
  | 'tr'
  | 'dk'
  | 'ua'
  | 'es-CO'
  | 'es-AR'
  | 'fr-FR'
  | 'es-ES'
  | 'ca'
  | 'en-SG'
  | 'ms-MY'
  | 'se'
```

File: app/locales/sk.json (hunk 2 above)
Comment (zacel) [reply app/locales/sk.json]: updated.. what a cool looking language!

File: app/locales/sk.json (hunk 3 above)
Comment (zacel) [reply app/locales/sk.json]: updated. thank you

**Comment by zacel** [comment]
@fossecode Maybe translations for things like this should be a separate pull request from the change it's self and then the main change, in this case the mobile banner, could be added after all the translations have been gathered and checked?

**Comment by adriaandotcom** [comment]
> I've changed the sk locale to use that one but wasn't able to find the duplicate line to remove it?

I removed the double linee, so if you updated the one line it's all good.

**Comment by fossecode** [comment]
> @fossecode Maybe translations for things like this should be a separate pull request from the change it's self and then the main change, in this case the mobile banner, could be added after all the translations have been gathered and checked?

Yeah, the hard part is gathering all the translations. Maybe we should create issues with label "translation" when we add new texts, and a checkbox for all the required languages. Then we could encourage people to go to the issue and fill in translations for their language.

**Comment by amritnagi** [comment]
But of a tangent but related... could we create some sort of command line script where you can specify the new string to add and it loops through all of the files and adds the string?
Could perhaps make it you can specify the French and Spanish translations and it assigns it to the relevant languages? (Could require a bit of config I think)

**Comment by zacel** [comment]
> > I've changed the sk locale to use that one but wasn't able to find the duplicate line to remove it?
> 
> I removed the double linee, so if you updated the one line it's all good.

Yes it should then be good to go. @fossecode so maybe I should remove the translations from this and then the pr can be just the banner with translations moved to an issue like `translations for mobile banner`? @amritnagi I think that sounds like a very good idea but someone like me might add a translation string and forget to change the other languages.

**Comment by bertrandlalo** [comment]
fr-FR
-----
- **"This page is not affiliated with the authorities."**: Cette page n'est pas affiliée aux autorités 
- **"Learn more!"**:  En savoir plus!

**Comment by tarasmatsyk** [comment]
UA:

```
  "This page is not affiliated with the authorities.":"Ця сторінка ніяк не зв'язана із владою.",
  "Learn more!":"Дізнатися більше!",
```

**Comment by rubentd** [comment]
`es-__`

"This page is not affiliated with the authorities.":"Esta página no está afiliada con las autoridades.",
  "Learn more!":"¡Conoce más!",

**Comment by adriaandotcom** [comment]
Closing this PR because I opened a new one with the fixed merge conflicts: https://github.com/BustByte/coronastatus/pull/475

**Pull request closed without merging**